taichi/ir/snode.h

```diff
@@ -92,14 +96,8 @@ class SNode {
 
   std::string get_node_type_name_hinted() const;
 
-  SNode();
-
-  SNode(int depth, SNodeType t);
-
-  ~SNode();
-
   SNode &insert_children(SNodeType t) {
-    ch.push_back(create(depth + 1, t));
+    ch.emplace_back(create(depth + 1, t));
```

Comment:
Actually `push_back` is probably better, see https://abseil.io/tips/112 (feel like i'm a language lawyer now..)

Reply:
Cool! Thanks for pointing this out - glad to learn something new about C++ everyday :-)